tests/testutil/service.go

```diff
@@ -302,6 +327,26 @@ func (ts *TestService) GetKubeCurrentWeight() int {
 	return getRevisionWeight(app, ts.Version)
 }
 
+// PodsResponsesMatchAvailableReplicas does a GetURL in the App endpoint and stores the response in a slice
+// the length of the resulting slice should represent the number of responsive pods in a service.
+// Returns true if the number of replicas is equal to the length of the responses slice.
+func (ts *TestService) PodsResponsesMatchAvailableReplicas(path string, numberOfReplicas int) bool {
```

Comment:
I don't understand this method, you'r hitting the app endpoint for the amount of ready replicas ? That could hit the same replica every time. Why not just hit each replica ? Whats the goal of this test ? We already test each service endpoint URL.

Reply:
Making sure the number of replicas match the different hostname responses from the `name.html` from each.

Reply:
If you're hitting app endpoint though, you could get the same replica every time. Even if you did split it out to hit different endpoints, only thing you're guaranteeing is that all pods are running.

Reply:
I think the app endpoint is balancing the requests properly I did some manual testing with curl in a service with scale=5 and I was getting the 5 different hostnames. Sometimes of course duplicated but eventually a did get them.

That's why I'm multiplying the scale by 5, to have room for probability of getting all them.

Reply:
You should be able to just get the individual pod endpoints and hit those. But even then it feels overkill to me, since this is all handled inside Kubernetes.

Reply:
But for validation we're also testing the balancer, this will be specially relevant for promotions: https://github.com/rancher/rio/issues/485

Reply:
Isn't that just balancing between revisions though ? Not the pods inside the revisions.

Reply:
I think that hits the latest promoted revision.

Reply:
I manually tested my previous comment and the App URL hits only the promoted revision when in 100% weight which is the case for these scenarios and only gets those app pods in the replicas. I think we would need separate requests methods when we deal with the weight/revision test cases.

Reply:
Ok. I still don't really like this approach, we can try it for now but if it gets flakey we'll need to redo. Entirely possible that if there is 5 replicas, 2 or 3 of them never get hit.

Reply:
That's a valid point but in theory if we have enough requests we should hit all the pods because the balancer works in Round Robin  (unless keep-alive by the browser or session caching(?) from kubernetes)

This has been working fine locally and in CI but you're right there's a chance we don't hit all the pods. I could work on a better solution if that becomes a problem.